backend/src/core/middlewares/auth.middleware.ts

```diff
@@ -219,11 +221,67 @@ export const InitAuthMiddleware = (authService: AuthService) => {
     }).catch((err) => next(err))
   }
 
+  /**
+   * Fetches sgID authorisation URL
+   * @param req
+   * @param res
+   */
+  const getSgidUrl = async (req: Request, res: Response): Promise<Response> => {
+    try {
+      const url = authService.getSgidUrl(req)
+      return res.status(200).send(url)
+    } catch (e) {
+      logger.error({
+        message: 'Error fetching sgID auth url',
+        error: e,
+      })
+      return res.sendStatus(500)
+    }
+  }
+
+  /**
+   * Verifies that the sgID code is valid
+   * @param req
+   * @param res
+   */
+  const verifySgidCode = async (
```

Comment:
nit: consider renaming the function to reflect that the point is to retrieve data from sgid. verifying that the sgid code is valid is a part of retrieving data from sgid but not vice-versa

Reply:
Updated naming to `verifySgidResponse`